src/Rocketeer/Strategies/Deploy/LocalCloneStrategy.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+namespace Rocketeer\Strategies\Deploy;
+use Rocketeer\Strategies\Deploy\SyncStrategy;
+use Rocketeer\Bash;
+
+class LocalCloneStrategy extends SyncStrategy
+{
+    /**
+     * @type string
+     */
+    protected $description = 'Uses rsync to create or update a release from local temporary cloned repository';
+    
+    /**
+     * Deploy a new clean copy of the application.
+     *
+     * @param string|null $destination
+     *
+     * @return bool
+     */
+    public function deploy($destination = null)
+    {
+        if (!$destination) {
+            $destination = $this->releasesManager->getCurrentReleasePath();
+        }
+        
+        // Create receiveing folder
+        $this->createFolder($destination);
+        
+        $tmpFolderPath = '.rocketeer/checkout/tmp/' . time() . '/';
+        $this->createCloneDirectory($tmpFolderPath);
+        
+        $this->cloneLocally($tmpFolderPath);
+        
+        return $this->rsyncTo($destination, $tmpFolderPath);
+    }
+    
+    /**
+     * Create clone directory if not exists
+     */
+    protected function createCloneDirectory($tmpFolderPath)
+    {
+        if (!is_dir($tmpFolderPath)) {
+            if (!mkdir($tmpFolderPath, 0777, true)) {
+                throw new \Exception("[" . __METHOD__ . "] Can't create clone directory : " . $tmpFolderPath);
+            }
+        }
```

Comment:
I check, it makeDirectory but it doesn't throw exception,I gonna use it but let the exception. It's OK ?